src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -127,6 +127,62 @@ public async Task<IActionResult> GetRecommendations(string sessionId)
             return Ok(output);
         }
 
+        /// <summary>
+        /// Gets the list of updatable option setting items for the selected recommendation.
+        /// </summary>
+        [HttpGet("session/<sessionId>/settings")]
+        [SwaggerOperation(OperationId = "GetConfigSettings")]
+        [SwaggerResponse(200, type: typeof(GetOptionSettingsOutput))]
+        [Authorize]
+        public IActionResult GetConfigSettings(string sessionId)
+        {
+            var state = _stateServer.Get(sessionId);
+            if (state == null)
+            {
+                return NotFound($"Session ID {sessionId} not found.");
+            }
+
+            if (state.SelectedRecommendation == null)
+            {
+                return NotFound($"A deployment target is not set for Session ID {sessionId}.");
+            }
+
+            var orchestrator = CreateOrchestrator(state);
+
+            var deploymentBundleDefinition = orchestrator.GetDeploymentBundleDefinition(state.SelectedRecommendation);
+
+            var configurableOptionSettings = state.SelectedRecommendation.Recipe.OptionSettings.Union(deploymentBundleDefinition.Parameters);
+
+            var output = new GetOptionSettingsOutput();
+
+            foreach (var setting in configurableOptionSettings)
+            {
+                if ((state.SelectedRecommendation.IsExistingCloudApplication && !setting.Updatable) || !state.SelectedRecommendation.IsOptionSettingDisplayable(setting))
```

Comment:
We should still send non updatable settings with a flag that says the setting is read only. That way the IDE can display all of the current settings for the deployment.

Reply:
Fixed